### PR TITLE
[api] Refactor /copy API with additional validation checks

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Add extra validation checks for the /copy API to catch edge case scenarios and improve error handling.

## How was this patch tested?

- Manually
- Unit tests

### CODE REVIEW

**Review by tabraiz12 — APPROVED** [review]
LGTM

File: apps/filebrowser/src/filebrowser/api.py [hunk 1]
```diff
@@ -656,17 +661,47 @@ def move(request):
...
+  if not request.fs.isdir(destination_path):
+    return HttpResponse('Destination path must be a directory.', status=400)
+
+  # Check if destination path is parent of source path
+  if _is_destination_parent_of_source(request, source_path, destination_path):
+    return HttpResponse('Destination cannot be the parent directory of source.', status=400)
+
+  # Check if file or folder already exists at destination path
+  if request.fs.exists(request.fs.join(destination_path, os.path.basename(source_path))):
+    return HttpResponse('File or folder already exists at destination path.', status=409)
```
Comment (JohanAhlen) [inline apps/filebrowser/src/filebrowser/api.py:698]: Possible to reuse the checks from move?

**Review by JohanAhlen — APPROVED** [review]
LGTM

**Review by tabraiz12 — APPROVED** [review]

File: apps/filebrowser/src/filebrowser/api.py (hunk 1 above)
Comment (Harshg999) [reply apps/filebrowser/src/filebrowser/api.py:698]: Good point, will reuse them once the move PR is merged.

File: apps/filebrowser/src/filebrowser/api.py [hunk 2]
```diff
@@ -642,6 +642,11 @@ def rename(request):
   return HttpResponse(status=200)
 
 
+def _is_destination_parent_of_source(request, source_path, destination_path):
```
[the remark this answers is not in the document]
Comment (Harshg999) [reply apps/filebrowser/src/filebrowser/api.py:645]: Good point, will reuse them once the move PR is merged.
